### PR TITLE
[Custom]: No way to associate class/constructor function.

### PR DESCRIPTION
**Pull request by dglazkov**
https://www.w3.org/Bugs/Public/show_bug.cgi?id=24019

### CODE REVIEW

**Comment by dglazkov** [comment]
@arv PTAL? Setting up the basics first.

File: spec/custom/index.html [hunk 1]
```diff
@@ -632,6 +633,81 @@ <h2 id="parsing">Parsing Custom Elements</h2>
...
+<dl>
+<dt>Input</dt>
+    <dd><var id="var-options-prototype">FUNCTION</var>, the custom element constructor function.</dd>
+    <dd><var>DOCUMENT</var>, the owner <a href="http://dom.spec.whatwg.org/#concept-document">document</a> for new <a href="#dfn-custom-element">custom element</a></dd>
+<dt>Output</dt>
+    <dd><var>FUNCTION</var>, the mutated custom element constructor function</dd>
+</dl>
+<ol>
+    <li>If <var>FUNCTION</var> is already an <a href="http://www.w3.org/TR/WebIDL/#dfn-interface-object">interface object</a> for any <a href="http://www.w3.org/TR/WebIDL/#dfn-interface">interface</a>, throw a <a href="http://dom.spec.whatwg.org/#notsupportederror"><code>NotSupportedError</code></a> and <strong>stop</strong>.</li>
+    <li>Let <var>DEFINITION</var> be an <a href="#dfn-element-definition">element definition</a> that has <var>FUNCTION</var> as <a href="#dfn-custom-element-prototype">custom element prototype</a></li>
```
Comment (arv) [inline spec/custom/index.html:694]: This is weird. FUNCITON is not the prototype. Maybe we need to use other terminology?

Comment (arv) [inline spec/custom/index.html:694]: Should probably be "custom element constructor function" instead?

**Comment by arv** [comment]
LGTM

File: spec/custom/index.html (hunk 1 above)
Comment (dglazkov) [inline spec/custom/index.html:694]: Right. This will be broken until I fix https://www.w3.org/Bugs/Public/show_bug.cgi?id=24020.

**Pull request merged**